docs/semgrep-secrets/historical-scanning.md

```diff
@@ -0,0 +1,114 @@
+1. Click **<i class="fa-solid fa-gear"></i> Settings**.
+1. Under Deployment, click the **<i class="fa-solid fa-toggle-large-on"></i> Historical scanning** toggle.
+![Historical scanning settings toggle](/img/historical-scanning-settings.png#md-width)
+
+Subsequent Semgrep full scans now include historical scanning.
+
+### Run a one-off historical scan
+
+To run a one-off or on-demand historical scan, you can create a specific CI job and then manually start the job as needed.
+
+The general steps are:
+
+1. Copy your current full scan CI job configuration file.
+1. Look for the `semgrep ci` command.
+1. Append the `--historical-secrets` flag:
+    ```
+    semgrep ci --historical-secrets
+    ```
+1. Depending on your CI provider, you may have to perform additional steps to enable the job to run manually. For example, GitHub Actions requires the `workflow_dispatch` event to be added to your CI job.
+
+
+### Run a local test scan
+
+You can run a historical scan locally without sending the scan results to Semgrep Cloud Platform.
+
+You can run this test to determine the time it takes for Semgrep Secrets to run on your repository's git commit history.
+
+To run a test scan, enter the following command:
+
+```bash
+semgrep scan --historical-secrets
+```
+
+The historical scan results appear in the **Secrets Historical Scan** section:
+
+![Historical scan section in the CLI](/img/historical-scans-cli.png#md-width)
+
+```
+## View or hide historical findings
+
+![Historical secrets in Semgrep Cloud Platform](/img/historical-secrets-scp.png)
+*Figure.* Historical findings in Semgrep Cloud Platform.
+
+1. Sign in to [<i class="fas fa-external-link fa-xs"></i> Semgrep Cloud Platform](https://semgrep.dev/login).
+1. Click **<i class="fa-solid fa-key"></i> Secrets**. Historical findings are identified by an **<i class="fa-solid fa-hourglass-half"></i>** icon.
+1. Click **<i class="fa-solid fa-hourglass-half"></i> Hide historical** to toggle the display of historical findings.
+
+## How it works
+
+The following sections discuss historical scan limitations, how findings are triaged, and other details that may affect how you implement historical scans in your security program.
+
+### Limitations
+
+- Historical scanning can slow down scan times. Depending on the size of your repository history, it can take as quickly as under 5 minutes to more than 60 minutes for extreme cases.
+- Within SCP, historical scan findings are not automatically marked as **Fixed**. Findings can only exist in two states: `Open` or `Ignored`. Because Semgrep scans do not automatically detect historical findings as fixed, you must manually rotate and triage the secret as `Ignored`.
+- A finding can show up twice in the CLI with historical scanning enabled: the HEAD commit in the regular Secrets scan and another commit in the historical scan.
+    - If findings are sent to SCP, they are deduplicated and appear as a **regular finding**, not a historical finding.
+
```

Comment:
Another limitation is that we only run regex-based Secrets rules, we don't run semantic rules. It also could be worth reiterating that we only create findings for valid secrets. no validator, invalid, or validation error are skipped

Reply:
> we don't run semantic rules

- Is the implication here, "no data-flow analysis, no constant propagation"? I suppose that's what makes it fast?
- By how much does that decrease our coverage? I'm under the impression a lot of our rules are semantic.

Reply:
You can get a sense of how many rules are regex vs. semantic from the [policies page](https://semgrep.dev/orgs/-/policies/secrets), but about 2/3 are regex. However this doesn't result in that much reduced coverage since each rule doesn't represent an equal number of findings.